site/content/en/docs/administration/advanced/analytics.md

```diff
@@ -154,6 +155,38 @@ Client events:
 
 <!--lint enable maximum-line-length-->
 
+### Working time calculation
+
+Here is a short overview of how CVAT deals with the user's working time:
+
+- The user interface collects events when a user interacts with the interface
+(resizing canvas, drawing objects, clicking buttons, etc)
+The structure of one single event is described [here](#events-log-structure).
+
+- The user interface sends these events in bulks to the server.
+Currently, it uses the following triggers to send events:
+    - Periodical timer (~90 seconds)
+    - A user clicks the "Save" button on the annotation view
+    - A user opens the annotation view
+    - A user closes the annotation view (but not the tab/browser)
+
+- When events reach the server, it calculates working time based on timestamps of the events.
+
+- Working time is computed as a sum of time differences between
+the end of the previous event and the end of the next event.
+    - But if the difference is more then threshold (~100 seconds), the difference is not added to the working time
+    - Regardless different events may have `duration` field, we consider only `change:frame`
+    event as a continuous event when computing working time
+    - So, the only `duration` of the `change:frame` event is added to the working time.
+    Thresholding is not involved here.
+
+- After calculation, the server generates `send:working_time` events with time value in payload.
+These events may be, or not be bounded to a certain job/task/project.
+CVAT defines belonging of the working time based on events, that were used to calculate.
+
+- If Clickhouse was configured, CVAT saves the event in
```

Comment:
Is it possible for Clickhouse to not be configured? I don't think it's optional.

Reply:
CVAT will work without grafana, clickhouse and vector. So, I think it is optional

Reply:
Let's omit this part, I do not think it is really important here